index.js

```diff
@@ -4,29 +4,35 @@ const exec = require('@actions/exec');
 const execa = require('execa');
 const { resolve } = require('path');
 
+const url = '403799762630.dkr.ecr.eu-west-1.amazonaws.com';
+
 function getEcrImage(deploy_file) {
   const deploy = require(resolve(deploy_file));
-  return deploy.taskDefinition
-      && deploy.taskDefinition.containerDefinitions
-      && deploy.taskDefinition.containerDefinitions[0].image || '403799762630.dkr.ecr.eu-west-1.amazonaws.com/' + deploy.service.serviceName;
+  return deploy.taskDefinition &&
+    deploy.taskDefinition.containerDefinitions &&
+    deploy.taskDefinition.containerDefinitions[0].image || `${url}/${deploy.service.serviceName}`;
 }
 
 async function tagDockerImage(current_tag, ecr_tag, ecr_tag_latest) {
   try {
     await exec.exec('docker', ['tag', current_tag, ecr_tag]);
     await exec.exec('docker', ['tag', current_tag, ecr_tag_latest]);
-  }
-  catch (e) {
+  } catch (e) {
     throw new Error('tagDockerImage failed');
   }
 }
 
 async function loginToECR() {
   try {
-    const { stdout: ecr_login_command } = await execa('aws', ['ecr', 'get-login', '--no-include-email']);
-    await exec.exec(ecr_login_command);
-  }
-  catch (e) {
+    const version_info = await execa('aws', ['--version']);
+    if (version_info.stdout.includes('aws-cli/2.')) {
+      const password = await execa('aws', ['ecr', 'get-login-password']);
+      await exec.exec('docker', ['login', '-u', 'AWS', '-p', password.stdout, url]);
```

Comment:
So the build machine has a command line tool with an aws cli. How does it understand where to get the password? Because it's already set up in an env var?

Reply:
The base-image used for the workflow has a `aws-cli` installed. We simply sign in. The CLI did change between version 1 and 2 though, and one difference is how you previously could get a command to sign in to docker using it.

The new way does not give you all that information.